File: run-migration.js
const { createClient } = require("@supabase/supabase-js");
const fs = require("fs");

const supabaseUrl = "https://ayohrbnesgstmjqbcbod.supabase.co";
const supabaseServiceKey = "sb_secret_dL8c32FrVjgvdXa3V6XMJw_5VATF0LU";

const supabase = createClient(supabaseUrl, supabaseServiceKey);

async function runMigration() {
  try {
    const sql = fs.readFileSync("./supabase-migration.sql", "utf8");
    const statements = sql
      .split(";")
      .map((s) => s.trim())
      .filter((s) => s && !s.startsWith("--"));

    for (const statement of statements) {
      console.log(`Executing: ${statement.substring(0, 50)}...`);
      const { error } = await supabase.rpc("exec", {
        command: statement,
      });

      if (error) {
        console.error("Error:", error);
      } else {
        console.log("✓ Success");
      }
    }

    console.log("\n✓ Migration complete!");
  } catch (err) {
    console.error("Migration failed:", err.message);
  }
}

runMigration();

File: run_migration.py
#!/usr/bin/env python3

import psycopg2
import sys

# Supabase connection details
conn_string = "host=db.ayohrbnesgstmjqbcbod.supabase.co port=5432 dbname=postgres user=postgres password='Apat3mp0@2026'"

try:
    conn = psycopg2.connect(conn_string)
    cursor = conn.cursor()
    print("✓ Connected to Supabase")

    # Read migration SQL
    with open('supabase-migration.sql', 'r') as f:
        sql = f.read()

    # Split by semicolon
    statements = [s.strip() for s in sql.split(';') if s.strip() and not s.strip().startswith('--')]

    # Execute each statement
    for stmt in statements:
        try:
            cursor.execute(stmt)
            print(f"✓ {stmt[:50]}...")
        except Exception as e:
            print(f"✗ {stmt[:50]}... - {str(e)[:60]}")

    conn.commit()
    print("\n✓ Migration complete!")
    cursor.close()
    conn.close()

except Exception as e:
    print(f"Error: {e}")
    sys.exit(1)
